### ISSUE TITLE
Jump condition for the Monte Carlo simulator 

### ISSUE THREAD

**Issue by i2000s** [qutip/qutip#111]
In the mcsolver function, it seems that (read from tutorial) the **quantum jumps** happen when a random number is generated larger than **(1-dp)**, where **dp** is no jump probability. And this is the only option for jumping in the algorithm. 

In theory, there is another strategy to choose **jump**--that is to use the concept of **waiting time** (see Dum and Zoller's PRA **45** (7), 1992 paper, for example). That is to say, we check the no jump time **t** to see if it is longer than the waiting time $\tau$: if **t** is longer, then a quantum jump happens; otherwise, there is no jump. 

Could anyone make the mcsolver flexible enough to let users freely to choose which strategy to use in the algorithm? Each of these two strategies have different meanings and their own advantages: the jumping probability strategy is good for simulating measurement models; the waiting time is more efficient if one just wants to know the averaged result. 

If currently none can implement this idea, could you provide a direction for me to write another mcsolver to use the counterpart strategy of jumping? Thanks.

Regards,
Qi

**Comment by goerz** [comment]
Is there some paper or other material that discusses the relative merits of the two approaches beyond numerical effort, either drawing a random number in each time step as in Mølmer at al, J. Opt. Soc. Am. B  10  524  (1993) or by choosing a wait time as in the Dum and Zoller Paper? At first glance, they don't seem to be mathematically equivalent to me. @i2000s, you mention that the former method is more appropriate for measurement models. Do you have a reference that explains this in more detail, by any chance?

**Comment by i2000s** [comment]
@goerz I think one of Mølmer's papers (maybe the one I have pointed out) addressed the equivalence of the two methods. Sorry, I don't have a definite paper to point out at this moment. I was actually organizing the Julia community to work on this direction (see [JuliaQuantum](http://juliaquantum.github.io)) from scratch, and hence didn't follow up the efforts in the measurement simulation within QuTiP. 

When I was writing my paper on continuous measurement ([PRA, 93.023817](http://journals.aps.org/pra/abstract/10.1103/PhysRevA.93.023817), I came up the argument that the former method may be more appropriate for measurement models. This is because measurement has to happen in evenly spaced time steps determined by the detector's integrating time, so it is intuitive to use a random number to judge if on each measurement step the quantum jump will happen. There may be a way to use the second method to implement the same measurement simulation model, but I didn't think too much on it. In reality, I developed another more efficient method without actually evolving the density operator in my paper to simulate continuous measurement using a set of stochastic equations for squeezing thousands of atoms' state which has not been included in QuTiP yet. Sorry I don't have more useful reference that explains this point in more details at this moment. Good luck!
